### PR TITLE
After trigger support

### LINKED ISSUE #215: Partition-friendly logging (triggers)
Logidze triggers are currently "BEFORE UPDATE OR INSERT"

If I were to try to [add logidze to a partitioned table](https://github.com/ankane/pgslice/issues/35) I'd get

> ActiveRecord::StatementInvalid: PG::WrongObjectType: 
> ERROR:  "device_location_events" is a partitioned table
> DETAIL:  Partitioned tables cannot have BEFORE / FOR EACH ROW triggers.

any concerns about me moving logidze to a "AFTER INSERT OR UPDATE" trigger on these tables?

### PR DESCRIPTION
**Pull request by palkan**
### What is the purpose of this pull request?

Closes #215. 

### What changes did you make? (overview)

This PR is a combination of #226 and #225 with some refactoring changes on top.

### Checklist

- [x] I've added tests for this change
- [x] I've added a Changelog entry
- [x] I've updated a documentation (Readme)

### CODE REVIEW

File: lib/generators/logidze/install/install_generator.rb [hunk 1]
```diff
@@ -79,6 +79,18 @@ def all_functions
...
           @function_definitions ||= Logidze::Utils::FunctionDefinitions.from_fs
         end
+
+        # Generate `logidze_logger_after.sql` from the regular `logidze_logger.sql`
+        # by find-and-replacing a few lines
+        def generate_logidze_logger_after
+          source = File.read(File.join(__dir__, "functions", "logidze_logger.sql"))
+          source.sub!(/^CREATE OR REPLACE FUNCTION logidze_logger.*$/, "")
+          source.sub!(/^  -- version.*$/, "")
+          source.sub!("  BEGIN", "BEGIN\n    IF pg_trigger_depth() > 1 THEN\n      RETURN NULL;\n    END IF;")
```
Comment (SparLaimor) [inline lib/generators/logidze/install/install_generator.rb:89]: Is there a functional difference between `pg_trigger_depth() > 1` and `WHEN (coalesce(current_setting('logidze.disabled', true), '') <> 'on' AND pg_trigger_depth() < 1)
` in `logidze_after.sql`?

Comment (palkan) [reply lib/generators/logidze/install/install_generator.rb:89]: I don't think there is. Let's move to the trigger.

File: README.md [hunk 2]
```diff
@@ -190,6 +191,18 @@ bundle exec rails destroy logidze:model Post
...
+
+Logidze supports partitioned tables for PostgreSQL 13+ without any additional configuration. For PostgreSQL 11/12, you should use _after_ triggers. To do that, provide the `--after-trigger` option to the migration:
+
+```sh
+bundle exec rails generate logidze:model Post --after-trigger
+```
+
+**NOTE:** Record changes are written as a full snapshot if the partition has changed during the update.
+
+**IMPORTANT:** Using Logidze for partitioned tables in PostgreSQL 10 is not supported.
```
Comment (prog-supdex) [inline README.md:204]: I guess, that here we can also say about 9.6 version
9.6 does not have declarative partitioned tables, but has inheritance and logidze also supports 9.6 pg version

Comment (palkan) [reply README.md:204]: Yeah, but we decided to discontinue PG <10 support since the next release.

**Comment by bf4** [comment]
THIS IS AMAZING ❤️ ❤️ ✨ 🦄 🕺

**Pull request merged**